code/datums/components/shell.dm

```diff
@@ -0,0 +1,178 @@
+/// Makes an atom a shell that is able to take in an attached circuit.
+/datum/component/shell
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+
+	/// The circuitboard attached to this shell
+	var/obj/item/integrated_circuit/attached_circuit
+
+	/// Flags containing what this shell can do
+	var/shell_flags = 0
+
+	/// The capacity of the shell.
+	var/capacity = INFINITY
+
+	/// A list of components that cannot be removed
+	var/list/obj/item/circuit_component/unremovable_components
```

Comment:
Just like before, I'm going to call here that every time you use "component" you should specify "circuit_component", because we don't want any confusion between these and DCS components.